packages/headless/src/controllers/insight/user-actions/headless-user-actions.ts

```diff
@@ -0,0 +1,145 @@
+import {UserActionTimeline} from '../../../api/service/insight/user-actions/user-actions-response';
+import {InsightEngine} from '../../../app/insight-engine/insight-engine';
+import {configuration, insightUserActions} from '../../../app/reducers';
+import {
+  fetchUserActions,
+  incrementNumberOfSessionsAfter,
+  incrementNumberOfSessionsBefore,
+  registerUserActions,
+} from '../../../features/insight-user-actions/insight-user-actions-actions';
+import {InsightAPIErrorStatusResponse} from '../../../insight.index';
+import {
+  ConfigurationSection,
+  InsightUserActionSection,
+} from '../../../state/state-sections';
+import {loadReducerError} from '../../../utils/errors';
+import {
+  buildController,
+  Controller,
+} from '../../controller/headless-controller';
+
+export interface UserActionsProps {
+  /**
+   * The options for the `UserActions` controller.
+   */
+  options: UserActionsOptions;
+}
+
+export interface UserActionsOptions {
+  /**
+   * The ticket creation date in the [ISO 8601](https://www.iso.org/iso-8601-date-and-time-format.html) format.
+   */
+  ticketCreationDate: string;
+  /**
+   * The number of sessions to include occuring before the ticket creation session.
+   * @defaultValue `0`
+   */
+  numberSessionsBefore?: number;
+  /**
+   * The number of sessions to include occuring after the ticket creation session.
+   * @defaultValue `0`
+   */
+  numberSessionsAfter?: number;
+  /**
+   * The names of custom events to exclude.
+   * @defaultValue `[]`
+   */
+  excludedCustomActions?: string[];
+}
+
+export interface UserActionsState {
+  /**
+   * The timeline of user actions.
+   */
+  timeline: UserActionTimeline;
```

Comment:
Follow up question, what will the state contain when for example we try to fetch **2 following sessions** but in reality there is only **one session** that has occurred after the creation of the ticket?

Reply:
The API (which is still WIP) will return the requested number of sessions up to the maximum number it finds. So if we request 2 preceding sessions but it only finds 1 or 0, then it will return 1 or 0.
I guess there could be a kind of paging system where the response contains the total number of sessions before and after to guide the client... 🤔. It would be more complex to implement but the ui could disable the buttons if it already knows there aren't any more sessions.